### PR TITLE
Add guards for tournament status when using commands

### PR DESCRIPTION
**Pull request by AlphaKretin**
## Description

Ensures a command cannot be used when the tournament is in an inappropriate status for that command. This also covers the registration of players.

## Checklist

- [x] I am following the [contributing guidelines]( https://github.com/AlphaKretin/emcee-tournament-bot/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: src/database/postgres.ts [hunk 1]
```diff
@@ -73,8 +78,30 @@ export class DatabaseWrapperPostgres implements DatabaseWrapper {
...
+
+	async getTournament(tournamentId: string, assertStatus?: TournamentStatus): Promise<DatabaseTournament> {
+		const tournament = await this.findTournament(tournamentId);
+		if (assertStatus && tournament.status !== assertStatus) {
+			throw new AssertStatusError(tournamentId, assertStatus, tournament.status);
+		}
+		return this.wrap(tournament);
 	}
 
 	async updateTournament(tournamentId: string, name: string, desc: string): Promise<void> {
```
Comment (kevinlul) [inline src/database/postgres.ts:107]: Worth changing the thrown exception to AssertStatusError here too?

File: src/util/errors.ts [hunk 2]
```diff
@@ -64,4 +66,16 @@ export class BlockedDMsError extends UserError {
 	}
 }
 
+export class AssertStatusError extends UserError {
+	tournamentId: string;
+	requiredStatus: TournamentStatus;
+	currentStatus: TournamentStatus;
+	constructor(tournamentId: string, requiredStatus: TournamentStatus, currentStatus: TournamentStatus) {
+		super(`Tournament ${tournamentId} must be in be ${requiredStatus}, but is currently ${currentStatus}.`);
```
Comment (kevinlul) [inline src/util/errors.ts:74]: typo

**Review by kevinlul — APPROVED** [review]
lol approving own code again

File: src/database/postgres.ts (hunk 1 above)
Comment (AlphaKretin) [reply src/database/postgres.ts:107]: Yes, I think updateTournament should use the same error

**Pull request merged**